packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -122,9 +122,25 @@ export class MediaPlaceholder extends Component {
 		let setMedia;
 		if ( multiple ) {
 			if ( addToGallery ) {
-				const currentValue = value;
+				let mediaToReturn = value;
 				setMedia = ( newMedia ) => {
```

Comment:
Hi @Addison-Stavlo,
Thank you for debugging these issues and proposing a solution to this problem 👍
This solution seems to work well.
I wonder if it would be possible to simplify the logic a bit by always containing the newMedia with this.props.value with all blob URLs filtered (removed) using isBlobUrl function in the filter from @wordpress/blob? Do you think the logic would be equivalent?

Reply:
Hi @jorgefilipecosta ,
Thanks for taking the time to give some feedback on this.  

> I wonder if it would be possible to simplify the logic a bit by always containing the newMedia with this.props.value with all blob URLs filtered (removed) using isBlobUrl function

It seems that this would remove blobs from other upload groups, add duplicates, and possibly introduce some other unwanted side effects that we would need to figure out how to get around.  However, we can use this idea of always concating `newMedia` onto a filtered `this.props.value` but instead filter out any media that corresponds to `newMedia` given by the last pass of the function.  

This should both clean up the logic a little bit as well as fix the remaining bug noted in example gif 3.